block/gs/adapter.go

```diff
@@ -121,6 +121,29 @@ func (a *Adapter) Get(obj block.ObjectPointer, _ int64) (io.ReadCloser, error) {
 	return r, err
 }
 
+func (a *Adapter) List(storageNamespace, prefix string) ([]string, error) {
```

Comment:
If we supply a listing interface the does a full scan, we will get a memory issue at some point.
Think that if we will accept a callback (with option return value to stop processing) in this case it cover the count use-case future requirements without hitting any memory issues.

Reply:
I prefer to have something like Walk or an Iterator and will use it later if needed